### ISSUE TITLE
5.0.1 fails to build wheel

### ISSUE THREAD

**Issue by stumpylog** [pikepdf/pikepdf#323]
I'm attempting to build pikepdf 5.0.1 for arm64 and armv7 architectures, for use Docker.

When I attempt to build pikepdf, using the following steps, it will fail due to something about the pyproject.toml file.

### Build Steps
1. git clone --quiet --depth 1 --branch v5.0.1 https://github.com/pikepdf/pikepdf.git
2. cd pikepdf
3. mkdir wheels
4. python3 -m pip wheel . -w wheels

### Error
```
configuration error: `project` must contain ['name'] properties
...
ValueError: invalid pyproject.toml config: `project`
```

Is there a specific `pip`, `setuptools` or `wheel` required to build?

**Comment by phillipsmjordan** [comment]
I started getting the same thing today. 

Not sure if related, but I took the OSX Monterey 12.3 update last night. Im running off of a `python-3.10` docker image

```
#14 24.34   EnvCommandError
#14 24.34 
#14 24.34   Command ['/usr/local/bin/python', '-m', 'pip', 'install', '--no-deps', '/root/.cache/pypoetry/artifacts/79/d5/ab/dfdd265662000b7e0642514acf04d3c75a8572824c62c222c71fff300a/pikepdf-5.0.1.tar.gz'] errored with the following return code 1, and output: 
#14 24.34   Processing /root/.cache/pypoetry/artifacts/79/d5/ab/dfdd265662000b7e0642514acf04d3c75a8572824c62c222c71fff300a/pikepdf-5.0.1.tar.gz
#14 24.34     Installing build dependencies: started
#14 24.34     Installing build dependencies: finished with status 'done'
#14 24.34     Getting requirements to build wheel: started
#14 24.34     Getting requirements to build wheel: finished with status 'done'
#14 24.34     Preparing metadata (pyproject.toml): started
#14 24.34     Preparing metadata (pyproject.toml): finished with status 'error'
#14 24.34     error: subprocess-exited-with-error
#14 24.34     
#14 24.34     × Preparing metadata (pyproject.toml) did not run successfully.
#14 24.34     │ exit code: 1
#14 24.34     ╰─> [515 lines of output]
#14 24.34         /tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:100: _ExperimentalProjectMetadata: Support for project metadata in `pyproject.toml` is still experimental and may be removed (or change) in future releases.
#14 24.34           warnings.warn(msg, _ExperimentalProjectMetadata)
#14 24.34         configuration error: `project` must contain ['name'] properties
#14 24.34         DESCRIPTION:
#14 24.34             Data structure for the **project** table inside ``pyproject.toml`` (as
#14 24.34             initially defined in :pep:`621`)
#14 24.34         
#14 24.34         GIVEN VALUE:
#14 24.34             {
#14 24.34                 "requires-python": ">=3.7"
#14 24.34             }
#14 24.34         
#14 24.34         OFFENDING RULE: 'required'
#14 24.34         
#14 24.34         DEFINITION:
#14 24.34             {
#14 24.34                 "$schema": "http://json-schema.org/draft-07/schema",
#14 24.34                 "$id": "https://packaging.python.org/en/latest/specifications/declaring-project-metadata/",
#14 24.34                 "title": "Package metadata stored in the ``project`` table",
#14 24.34                 "type": "object",
#14 24.34                 "properties": {
#14 24.34                     "name": {
#14 24.34                         "type": "string",
#14 24.34                         "description": "The name (primary identifier) of the project. MUST be statically defined.",
#14 24.34                         "format": "pep508-identifier"
#14 24.34                     },
#14 24.34                     "version": {
#14 24.34                         "type": "string",
#14 24.34                         "description": "The version of the project as supported by :pep:`440`.",
#14 24.34                         "format": "pep440"
#14 24.34                     },
#14 24.34                     "description": {
#14 24.34                         "type": "string",
#14 24.34                         "$$description": [
#14 24.34                             "The `summary description of the project",
#14 24.34                             "<https://packaging.python.org/specifications/core-metadata/#summary>`_"
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "readme": {
#14 24.34                         "$$description": [
#14 24.34                             "`Full/detailed description of the project in the form of a README",
#14 24.34                             "<https://www.python.org/dev/peps/pep-0621/#readme>`_",
#14 24.34                             "with meaning similar to the one defined in `core metadata's Description",
#14 24.34                             "<https://packaging.python.org/specifications/core-metadata/#description>`_"
#14 24.34                         ],
#14 24.34                         "oneOf": [
#14 24.34                             {
#14 24.34                                 "type": "string",
#14 24.34                                 "$$description": [
#14 24.34                                     "Relative path to a text file (UTF-8) containing the full description",
#14 24.34                                     "of the project. If the file path ends in case-insensitive ``.md`` or",
#14 24.34                                     "``.rst`` suffixes, then the content-type is respectively",
#14 24.34                                     "``text/markdown`` or ``text/x-rst``"
#14 24.34                                 ]
#14 24.34                             },
#14 24.34                             {
#14 24.34                                 "type": "object",
#14 24.34                                 "allOf": [
#14 24.34                                     {
#14 24.34                                         "anyOf": [
#14 24.34                                             {
#14 24.34                                                 "properties": {
#14 24.34                                                     "file": {
#14 24.34                                                         "type": "string",
#14 24.34                                                         "$$description": [
#14 24.34                                                             "Relative path to a text file containing the full description",
#14 24.34                                                             "of the project."
#14 24.34                                                         ]
#14 24.34                                                     }
#14 24.34                                                 },
#14 24.34                                                 "required": [
#14 24.34                                                     "file"
#14 24.34                                                 ]
#14 24.34                                             },
#14 24.34                                             {
#14 24.34                                                 "properties": {
#14 24.34                                                     "text": {
#14 24.34                                                         "type": "string",
#14 24.34                                                         "description": "Full text describing the project."
#14 24.34                                                     }
#14 24.34                                                 },
#14 24.34                                                 "required": [
#14 24.34                                                     "text"
#14 24.34                                                 ]
#14 24.34                                             }
#14 24.34                                         ]
#14 24.34                                     },
#14 24.34                                     {
#14 24.34                                         "properties": {
#14 24.34                                             "content-type": {
#14 24.34                                                 "type": "string",
#14 24.34                                                 "$$description": [
#14 24.34                                                     "Content-type (:rfc:`1341`) of the full description",
#14 24.34                                                     "(e.g. ``text/markdown``). The ``charset`` parameter is assumed",
#14 24.34                                                     "UTF-8 when not present."
#14 24.34                                                 ],
#14 24.34                                                 "$comment": "TODO: add regex pattern or format?"
#14 24.34                                             }
#14 24.34                                         },
#14 24.34                                         "required": [
#14 24.34                                             "content-type"
#14 24.34                                         ]
#14 24.34                                     }
#14 24.34                                 ]
#14 24.34                             }
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "requires-python": {
#14 24.34                         "type": "string",
#14 24.34                         "format": "pep508-versionspec",
#14 24.34                         "$$description": [
#14 24.34                             "`The Python version requirements of the project",
#14 24.34                             "<https://packaging.python.org/specifications/core-metadata/#requires-python>`_."
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "license": {
#14 24.34                         "description": "`Project license <https://www.python.org/dev/peps/pep-0621/#license>`_.",
#14 24.34                         "oneOf": [
#14 24.34                             {
#14 24.34                                 "properties": {
#14 24.34                                     "file": {
#14 24.34                                         "type": "string",
#14 24.34                                         "$$description": [
#14 24.34                                             "Relative path to the file (UTF-8) which contains the license for the",
#14 24.34                                             "project."
#14 24.34                                         ]
#14 24.34                                     }
#14 24.34                                 },
#14 24.34                                 "required": [
#14 24.34                                     "file"
#14 24.34                                 ]
#14 24.34                             },
#14 24.34                             {
#14 24.34                                 "properties": {
#14 24.34                                     "text": {
#14 24.34                                         "type": "string",
#14 24.34                                         "$$description": [
#14 24.34                                             "The license of the project whose meaning is that of the",
#14 24.34                                             "`License field from the core metadata",
#14 24.34                                             "<https://packaging.python.org/specifications/core-metadata/#license>`_."
#14 24.34                                         ]
#14 24.34                                     }
#14 24.34                                 },
#14 24.34                                 "required": [
#14 24.34                                     "text"
#14 24.34                                 ]
#14 24.34                             }
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "authors": {
#14 24.34                         "type": "array",
#14 24.34                         "items": {
#14 24.34                             "$id": "#/definitions/author",
#14 24.34                             "title": "Author or Maintainer",
#14 24.34                             "$comment": "https://www.python.org/dev/peps/pep-0621/#authors-maintainers",
#14 24.34                             "type": "object",
#14 24.34                             "properties": {
#14 24.34                                 "name": {
#14 24.34                                     "type": "string",
#14 24.34                                     "$$description": [
#14 24.34                                         "MUST be a valid email name, i.e. whatever can be put as a name, before an",
#14 24.34                                         "email, in :rfc:`822`."
#14 24.34                                     ]
#14 24.34                                 },
#14 24.34                                 "email": {
#14 24.34                                     "type": "string",
#14 24.34                                     "format": "idn-email",
#14 24.34                                     "description": "MUST be a valid email address"
#14 24.34                                 }
#14 24.34                             }
#14 24.34                         },
#14 24.34                         "$$description": [
#14 24.34                             "The people or organizations considered to be the 'authors' of the project.",
#14 24.34                             "The exact meaning is open to interpretation (e.g. original or primary authors,",
#14 24.34                             "current maintainers, or owners of the package)."
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "maintainers": {
#14 24.34                         "type": "array",
#14 24.34                         "items": {
#14 24.34                             "$id": "#/definitions/author",
#14 24.34                             "title": "Author or Maintainer",
#14 24.34                             "$comment": "https://www.python.org/dev/peps/pep-0621/#authors-maintainers",
#14 24.34                             "type": "object",
#14 24.34                             "properties": {
#14 24.34                                 "name": {
#14 24.34                                     "type": "string",
#14 24.34                                     "$$description": [
#14 24.34                                         "MUST be a valid email name, i.e. whatever can be put as a name, before an",
#14 24.34                                         "email, in :rfc:`822`."
#14 24.34                                     ]
#14 24.34                                 },
#14 24.34                                 "email": {
#14 24.34                                     "type": "string",
#14 24.34                                     "format": "idn-email",
#14 24.34                                     "description": "MUST be a valid email address"
#14 24.34                                 }
#14 24.34                             }
#14 24.34                         },
#14 24.34                         "$$description": [
#14 24.34                             "The people or organizations considered to be the 'maintainers' of the project.",
#14 24.34                             "Similarly to ``authors``, the exact meaning is open to interpretation."
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "keywords": {
#14 24.34                         "type": "array",
#14 24.34                         "items": {
#14 24.34                             "type": "string"
#14 24.34                         },
#14 24.34                         "description": "List of keywords to assist searching for the distribution in a larger catalog."
#14 24.34                     },
#14 24.34                     "classifiers": {
#14 24.34                         "type": "array",
#14 24.34                         "items": {
#14 24.34                             "type": "string",
#14 24.34                             "format": "trove-classifier",
#14 24.34                             "description": "`PyPI classifier <https://pypi.org/classifiers/>`_."
#14 24.34                         },
#14 24.34                         "$$description": [
#14 24.34                             "`Trove classifiers <https://pypi.org/classifiers/>`_",
#14 24.34                             "which apply to the project."
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "urls": {
#14 24.34                         "type": "object",
#14 24.34                         "description": "URLs associated with the project in the form ``label => value``.",
#14 24.34                         "additionalProperties": false,
#14 24.34                         "patternProperties": {
#14 24.34                             "^.+$": {
#14 24.34                                 "type": "string",
#14 24.34                                 "format": "url"
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "scripts": {
#14 24.34                         "$id": "#/definitions/entry-point-group",
#14 24.34                         "title": "Entry-points",
#14 24.34                         "type": "object",
#14 24.34                         "$$description": [
#14 24.34                             "Entry-points are grouped together to indicate what sort of capabilities they",
#14 24.34                             "provide.",
#14 24.34                             "See the `packaging guides",
#14 24.34                             "<https://packaging.python.org/specifications/entry-points/>`_",
#14 24.34                             "and `setuptools docs",
#14 24.34                             "<https://setuptools.pypa.io/en/latest/userguide/entry_point.html>`_",
#14 24.34                             "for more information."
#14 24.34                         ],
#14 24.34                         "propertyNames": {
#14 24.34                             "format": "python-entrypoint-name"
#14 24.34                         },
#14 24.34                         "additionalProperties": false,
#14 24.34                         "patternProperties": {
#14 24.34                             "^.+$": {
#14 24.34                                 "type": "string",
#14 24.34                                 "$$description": [
#14 24.34                                     "Reference to a Python object. It is either in the form",
#14 24.34                                     "``importable.module``, or ``importable.module:object.attr``."
#14 24.34                                 ],
#14 24.34                                 "format": "python-entrypoint-reference",
#14 24.34                                 "$comment": "https://packaging.python.org/specifications/entry-points/"
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "gui-scripts": {
#14 24.34                         "$id": "#/definitions/entry-point-group",
#14 24.34                         "title": "Entry-points",
#14 24.34                         "type": "object",
#14 24.34                         "$$description": [
#14 24.34                             "Entry-points are grouped together to indicate what sort of capabilities they",
#14 24.34                             "provide.",
#14 24.34                             "See the `packaging guides",
#14 24.34                             "<https://packaging.python.org/specifications/entry-points/>`_",
#14 24.34                             "and `setuptools docs",
#14 24.34                             "<https://setuptools.pypa.io/en/latest/userguide/entry_point.html>`_",
#14 24.34                             "for more information."
#14 24.34                         ],
#14 24.34                         "propertyNames": {
#14 24.34                             "format": "python-entrypoint-name"
#14 24.34                         },
#14 24.34                         "additionalProperties": false,
#14 24.34                         "patternProperties": {
#14 24.34                             "^.+$": {
#14 24.34                                 "type": "string",
#14 24.34                                 "$$description": [
#14 24.34                                     "Reference to a Python object. It is either in the form",
#14 24.34                                     "``importable.module``, or ``importable.module:object.attr``."
#14 24.34                                 ],
#14 24.34                                 "format": "python-entrypoint-reference",
#14 24.34                                 "$comment": "https://packaging.python.org/specifications/entry-points/"
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "entry-points": {
#14 24.34                         "$$description": [
#14 24.34                             "Instruct the installer to expose the given modules/functions via",
#14 24.34                             "``entry-point`` discovery mechanism (useful for plugins).",
#14 24.34                             "More information available in the `Python packaging guide",
#14 24.34                             "<https://packaging.python.org/specifications/entry-points/>`_."
#14 24.34                         ],
#14 24.34                         "propertyNames": {
#14 24.34                             "format": "python-entrypoint-group"
#14 24.34                         },
#14 24.34                         "additionalProperties": false,
#14 24.34                         "patternProperties": {
#14 24.34                             "^.+$": {
#14 24.34                                 "$id": "#/definitions/entry-point-group",
#14 24.34                                 "title": "Entry-points",
#14 24.34                                 "type": "object",
#14 24.34                                 "$$description": [
#14 24.34                                     "Entry-points are grouped together to indicate what sort of capabilities they",
#14 24.34                                     "provide.",
#14 24.34                                     "See the `packaging guides",
#14 24.34                                     "<https://packaging.python.org/specifications/entry-points/>`_",
#14 24.34                                     "and `setuptools docs",
#14 24.34                                     "<https://setuptools.pypa.io/en/latest/userguide/entry_point.html>`_",
#14 24.34                                     "for more information."
#14 24.34                                 ],
#14 24.34                                 "propertyNames": {
#14 24.34                                     "format": "python-entrypoint-name"
#14 24.34                                 },
#14 24.34                                 "additionalProperties": false,
#14 24.34                                 "patternProperties": {
#14 24.34                                     "^.+$": {
#14 24.34                                         "type": "string",
#14 24.34                                         "$$description": [
#14 24.34                                             "Reference to a Python object. It is either in the form",
#14 24.34                                             "``importable.module``, or ``importable.module:object.attr``."
#14 24.34                                         ],
#14 24.34                                         "format": "python-entrypoint-reference",
#14 24.34                                         "$comment": "https://packaging.python.org/specifications/entry-points/"
#14 24.34                                     }
#14 24.34                                 }
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "dependencies": {
#14 24.34                         "type": "array",
#14 24.34                         "description": "Project (mandatory) dependencies.",
#14 24.34                         "items": {
#14 24.34                             "$id": "#/definitions/dependency",
#14 24.34                             "title": "Dependency",
#14 24.34                             "type": "string",
#14 24.34                             "description": "Project dependency specification according to PEP 508",
#14 24.34                             "format": "pep508"
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "optional-dependencies": {
#14 24.34                         "type": "object",
#14 24.34                         "description": "Optional dependency for the project",
#14 24.34                         "propertyNames": {
#14 24.34                             "format": "pep508-identifier"
#14 24.34                         },
#14 24.34                         "additionalProperties": false,
#14 24.34                         "patternProperties": {
#14 24.34                             "^.+$": {
#14 24.34                                 "type": "array",
#14 24.34                                 "items": {
#14 24.34                                     "$id": "#/definitions/dependency",
#14 24.34                                     "title": "Dependency",
#14 24.34                                     "type": "string",
#14 24.34                                     "description": "Project dependency specification according to PEP 508",
#14 24.34                                     "format": "pep508"
#14 24.34                                 }
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "dynamic": {
#14 24.34                         "type": "array",
#14 24.34                         "$$description": [
#14 24.34                             "Specifies which fields are intentionally unspecified and expected to be",
#14 24.34                             "dynamically provided by build tools"
#14 24.34                         ],
#14 24.34                         "items": {
#14 24.34                             "enum": [
#14 24.34                                 "version",
#14 24.34                                 "description",
#14 24.34                                 "readme",
#14 24.34                                 "requires-python",
#14 24.34                                 "license",
#14 24.34                                 "authors",
#14 24.34                                 "maintainers",
#14 24.34                                 "keywords",
#14 24.34                                 "classifiers",
#14 24.34                                 "urls",
#14 24.34                                 "scripts",
#14 24.34                                 "gui-scripts",
#14 24.34                                 "entry-points",
#14 24.34                                 "dependencies",
#14 24.34                                 "optional-dependencies"
#14 24.34                             ]
#14 24.34                         }
#14 24.34                     }
#14 24.34                 },
#14 24.34                 "required": [
#14 24.34                     "name"
#14 24.34                 ],
#14 24.34                 "additionalProperties": false,
#14 24.34                 "if": {
#14 24.34                     "not": {
#14 24.34                         "required": [
#14 24.34                             "version"
#14 24.34                         ],
#14 24.34                         "$$description": [
#14 24.34                             "version is statically defined in the ``version`` field"
#14 24.34                         ]
#14 24.34                     },
#14 24.34                     "$$comment": [
#14 24.34                         "According to :pep:`621`:",
#14 24.34                         "    If the core metadata specification lists a field as \"Required\", then",
#14 24.34                         "    the metadata MUST specify the field statically or list it in dynamic",
#14 24.34                         "In turn, `core metadata`_ defines:",
#14 24.34                         "    The required fields are: Metadata-Version, Name, Version.",
#14 24.34                         "    All the other fields are optional.",
#14 24.34                         "Since ``Metadata-Version`` is defined by the build back-end, ``name`` and",
#14 24.34                         "``version`` are the only mandatory information in ``pyproject.toml``.",
#14 24.34                         ".. _core metadata: https://packaging.python.org/specifications/core-metadata/"
#14 24.34                     ]
#14 24.34                 },
#14 24.34                 "then": {
#14 24.34                     "properties": {
#14 24.34                         "dynamic": {
#14 24.34                             "contains": {
#14 24.34                                 "const": "version"
#14 24.34                             },
#14 24.34                             "$$description": [
#14 24.34                                 "version should be listed in ``dynamic``"
#14 24.34                             ]
#14 24.34                         }
#14 24.34                     }
#14 24.34                 },
#14 24.34                 "definitions": {
#14 24.34                     "author": {
#14 24.34                         "$id": "#/definitions/author",
#14 24.34                         "title": "Author or Maintainer",
#14 24.34                         "$comment": "https://www.python.org/dev/peps/pep-0621/#authors-maintainers",
#14 24.34                         "type": "object",
#14 24.34                         "properties": {
#14 24.34                             "name": {
#14 24.34                                 "type": "string",
#14 24.34                                 "$$description": [
#14 24.34                                     "MUST be a valid email name, i.e. whatever can be put as a name, before an",
#14 24.34                                     "email, in :rfc:`822`."
#14 24.34                                 ]
#14 24.34                             },
#14 24.34                             "email": {
#14 24.34                                 "type": "string",
#14 24.34                                 "format": "idn-email",
#14 24.34                                 "description": "MUST be a valid email address"
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "entry-point-group": {
#14 24.34                         "$id": "#/definitions/entry-point-group",
#14 24.34                         "title": "Entry-points",
#14 24.34                         "type": "object",
#14 24.34                         "$$description": [
#14 24.34                             "Entry-points are grouped together to indicate what sort of capabilities they",
#14 24.34                             "provide.",
#14 24.34                             "See the `packaging guides",
#14 24.34                             "<https://packaging.python.org/specifications/entry-points/>`_",
#14 24.34                             "and `setuptools docs",
#14 24.34                             "<https://setuptools.pypa.io/en/latest/userguide/entry_point.html>`_",
#14 24.34                             "for more information."
#14 24.34                         ],
#14 24.34                         "propertyNames": {
#14 24.34                             "format": "python-entrypoint-name"
#14 24.34                         },
#14 24.34                         "additionalProperties": false,
#14 24.34                         "patternProperties": {
#14 24.34                             "^.+$": {
#14 24.34                                 "type": "string",
#14 24.34                                 "$$description": [
#14 24.34                                     "Reference to a Python object. It is either in the form",
#14 24.34                                     "``importable.module``, or ``importable.module:object.attr``."
#14 24.34                                 ],
#14 24.34                                 "format": "python-entrypoint-reference",
#14 24.34                                 "$comment": "https://packaging.python.org/specifications/entry-points/"
#14 24.34                             }
#14 24.34                         }
#14 24.34                     },
#14 24.34                     "dependency": {
#14 24.34                         "$id": "#/definitions/dependency",
#14 24.34                         "title": "Dependency",
#14 24.34                         "type": "string",
#14 24.34                         "description": "Project dependency specification according to PEP 508",
#14 24.34                         "format": "pep508"
#14 24.34                     }
#14 24.34                 }
#14 24.34             }
#14 24.34         Traceback (most recent call last):
#14 24.34           File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
#14 24.34             main()
#14 24.34           File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
#14 24.34             json_out['return_val'] = hook(**hook_input['kwargs'])
#14 24.34           File "/usr/local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
#14 24.34             return hook(metadata_directory, config_settings)
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 188, in prepare_metadata_for_build_wheel
#14 24.34             self.run_setup()
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 174, in run_setup
#14 24.34             exec(compile(code, __file__, 'exec'), locals())
#14 24.34           File "setup.py", line 66, in <module>
#14 24.34             setup(
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/__init__.py", line 77, in setup
#14 24.34             return distutils.core.setup(**attrs)
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 122, in setup
#14 24.34             dist.parse_config_files()
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/dist.py", line 836, in parse_config_files
#14 24.34             pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 52, in apply_configuration
#14 24.34             config = read_configuration(filepath, True, ignore_option_errors, dist)
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 114, in read_configuration
#14 24.34             validate(subset, filepath)
#14 24.34           File "/tmp/pip-build-env-8utwm61s/overlay/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 43, in validate
#14 24.34             raise error from None
#14 24.34         ValueError: invalid pyproject.toml config: `project`
#14 24.34         [end of output]
```

**Comment by jbarlow83** [comment]
I won't have time to look at any issues till April at the earliest. Please feel free investigate and submit PRs if you need something done before then.

**Comment by phillipsmjordan** [comment]
Other folks reporting similar issues in the `setuptools` repo https://github.com/pypa/setuptools/issues/3199

**Comment by phillipsmjordan** [comment]
I'm getting a 403 when trying to push to this repo, but this commit fixed it for me https://github.com/soap-bx/pikepdf/commit/a28386f79d19bb587fb685a5c86e86a0cb8fd999

**Comment by mara004** [comment]
That's because of how GitHub works. You can't directly push to a repository unless you have write access. Fork pikepdf and make a Pull Request instead.

**Comment by phillipsmjordan** [comment]
> @phillipsmjordan That's because of how GitHub works. You can't directly push to a repository unless you have write access, of course. Fork pikepdf and make a Pull Request instead (just click on `Contribute` → `Open pull request`, or use the GitHub CLI with `gh pr create`).

Ah thanks. I've been on Gitlab for years now and have lost all touch with reality. 

https://github.com/pikepdf/pikepdf/pull/324

**Comment by johncronan** [comment]
Does anyone here know how I can fix my docker build, in the meantime? Where does the "setuptools<=60.10.0" go? The only thing I've managed so far is that it succeeds if I remove the dependency entirely.

**Comment by stumpylog** [comment]
My quick fix was removing `pyproject.toml` entirely.  There didn't appear to be anything in it for my purpose of building a wheel for various Arm architectures

**Comment by johncronan** [comment]
I use `poetry install.` Then the underlying `pip install` fails. But only on ARM.

**Comment by jrast** [comment]
Since #324 was merged, pikepdf can be installed with `pip install git+https://github.com/pikepdf/pikepdf.git#egg=pikepdf`

**Comment by phillipsmjordan** [comment]
> I use `poetry install`. Then the underlying `pip install` fails. But only on ARM.

What error?

**Comment by johncronan** [comment]
Same output as you had, with 

```
configuration error: `project` must contain ['name'] properties
```

**Comment by johncronan** [comment]
I did a `brew install qpdf` for my local environment (since I'm not just loading the wheel there, now) and put this in poetry.dependencies:

`pikepdf = { git = "https://github.com/pikepdf/pikepdf.git" }`

And with git added into the install packages for the docker build, it's going again. Thanks, everyone.

**Comment by jbarlow83** [comment]
Fixed in 5.1.1
